### PR TITLE
doc: release-notes: Add 3.7.0 release notes for networking

### PR DESCRIPTION
**Pull request by rlubos**
Add 3.7.0 release notes for networking (part of it).

### CODE REVIEW

File: doc/releases/release-notes-3.7.rst [hunk 1]
```diff
@@ -536,42 +536,238 @@ Drivers and Sensors
...
+* MQTT:
+
+  * Added ALPN support for MQTT TLS backend.
+  * Added user data field in :c:struct:`mqtt_client` context structure.
+
+* Network Interface:
+
+  * Added new API functions:
+
+    * :c:func:`net_if_ipv4_maddr_foreach
```
Comment (pdgendt) [inline doc/releases/release-notes-3.7.rst:701]: Missing backtick

```suggestion
    * :c:func:`net_if_ipv4_maddr_foreach`
```

Comment (rlubos) [reply doc/releases/release-notes-3.7.rst]: Fixed

**Review by pdgendt — APPROVED** [review]

File: doc/releases/release-notes-3.7.rst [hunk 2]
```diff
@@ -536,42 +536,238 @@ Drivers and Sensors
...
     default, if :kconfig:option:`CONFIG_NET_CONFIG_CLOCK_SNTP_INIT` is enabled.
+  * The syslog server address can now be set with DHCPv4 option. This is done by
+    default, if :kconfig:option:`CONFIG_LOG_BACKEND_NET_USE_DHCPV4_OPTION` is enabled.
+  * Fixed a bug, where options with registered callbacks were not requested from
+    the server.
+  * Fixed a bug, where netmask received from the server was not applied correctly.
+  * Reimplemented DHCPv4 client RENEW/REBIND logic to be compliant with RFC2131.
+  * Improved declined addresses management in DHCPv4 server, which now can be
+    reused after configured time.
+  * Fixed including client ID option in DHCPv4 server responsed, according to RFC6842.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:584]:
```suggestion
  * Fixed including the client ID option in the DHCPv4 server response, according to RFC6842.
```
?

File: doc/releases/release-notes-3.7.rst [hunk 3]
```diff
@@ -536,42 +536,238 @@ Drivers and Sensors
...
+  * Added structured logging support to syslog net backend.
+  * Added TCP support to syslog net backend.
 
 * TCP:
 
+  * Fixed possible deadlock when accepting new TCP connection.
+  * Fixed ACK number verification during connection teardown.
+  * Fixed a bug, where data bytes included in FIN packet were ignored.
+  * Fixed a possible TCP context leak in case initial SYN packet transmission failed.
+  * Deprecated ``CONFIG_NET_TCP_ACK_TIMEOUT`` as it was redundant with other configs.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:746]:
```suggestion
  * Deprecated :kconfig:option:`CONFIG_NET_TCP_ACK_TIMEOUT` as it was redundant with other configs.
```

**Review by aescolar — APPROVED** [review]
Thanks @rlubos 
There is a few articles missing in general,
and 2 other minor things, but it is good overall.
If you need to push an update you are welcome to fix them. Otherwise we can just fix that when we do the final polish of the release notes.

**Pull request merged**